lib/logstash/inputs/neo4j.rb

```diff
@@ -0,0 +1,87 @@
+# encoding: utf-8
+require "logstash/inputs/base"
+require "logstash/namespace"
+require "logstash/inputs/neo4j-client"
+
+# This plugin gets data from a Neo4j database in predefined intervals. To fetch
+# this data uses a given Cypher query.
+#
+# ### Usage:
+# [source, ruby]
+# input {
+#   neo4j {
+#     query => "MATCH (p:`Person`)-->(m:`Movie`) WHERE m.released = 2005 RETURN *"
+#     path  => "/foo/bar.db"
+#   }
+# }
+#
+# In embedded_db mode this plugin require a neo4j db 2.0.1 or superior. If
+# using the remote version there is no major restriction.
+#
+class LogStash::Inputs::Neo4j < LogStash::Inputs::Base
+
+  config_name "neo4j"
+
+  # If undefined, Logstash will complain, even if codec is unused.
+  default :codec, "plain"
+
+  # Cypher query used to retrieve data from the neo4j database, this statement
+  # should looks like something like this:
+  #
+  # MATCH (p:`Person`)-->(m:`Movie`) WHERE m.released = 2005 RETURN *
+  #
+  config :query, :validate => :string, :required => true
+
+  # The path within your file system where the neo4j database is located
+  config :path, :validate => :string, :required => true
+
+  # Schedule of when to periodically run statement, in Cron format
+  # for example: "* * * * *" (execute query every minute, on the minute)
+  config :schedule, :validate => :string
+
+  public
+  def register
+    require "rufus/scheduler"
+    require "logstash/inputs/neo4j-client"
+    @client = Neo4jrb::Client.open(@path)
+  end # def register
+
+  def run(queue)
+    if @schedule
+      schedule(queue)
+    else
+      fetch(queue)
```

Comment:
I tend to avoid one liners like this as I find them more complicated to read for some people. But yes @jsvd I agree this should be better documented.